schema/crdb/46.0.0/up3.sql

```diff
@@ -0,0 +1,3 @@
+-- We are dropping `kind` so that we can drop the `sled_resource` kind; we'll
+-- then recreate it (with some variants removed) and add this column back.
+ALTER TABLE omicron.public.sled_resource DROP COLUMN IF EXISTS kind;
```

Comment:
Hmm, I think this needs to be in its own schema version (e.g. 46.0.1).

If `up3.sql` happens, then the migration fails, we'll attempt to repeat `up2.sql` -- which will fail forever.